lib/auth/methods.go

```diff
@@ -126,61 +137,99 @@ func (s *Server) AuthenticateUser(ctx context.Context, req AuthenticateUserReque
 		username = actualUsername
 	}
 
+	user, err := s.GetUser(ctx, username, false /* withSecrets */)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	// After we're sure that the user has been logged in successfully, we should call
+	// the registered login hooks. Login hooks can be registered by other processes to
+	// execute arbitrary operations after a successful login.
+	if err := s.CallLoginHooks(ctx, user); err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	userState, err := s.GetUserOrLoginState(ctx, user.GetName())
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	clusterName, err := s.GetClusterName()
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+	accessInfo := services.AccessInfoFromUserState(userState)
+	checker, err := services.NewAccessChecker(accessInfo, clusterName.GetClusterName(), s)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
```

Comment:
Was it a bug that we weren't doing this check in `AuthenticateWebUser` previously?
https://github.com/gravitational/teleport/blob/70de43e913c0ec10c8d3e55d1cbceefa68a6d87b/lib/auth/methods.go#L490

Separate but related: generally it strikes me that this function is not very well tested (afaict) despite being load bearing for system security.

Reply:
That we weren't getting the `checker` inside the function? It wasn't a bug, we were just getting it after the function call. This change is just a refactor.

I think most paths are covered by `auth_login_test.go`.

Reply:
Yeah my question makes no sense, we aren't even using the checker to check anything here. My mistake.